src/nvim/path.c

```diff
@@ -1267,13 +1250,12 @@ int gen_expand_wildcards(int num_pat, char **pat, int *num_file, char ***file, i
         if (p == NULL) {
           p = (char_u *)pat[i];
         }
+      } else {
 #ifdef UNIX
-        /*
-         * On Unix, if expand_env() can't expand an environment
-         * variable, use the shell to do that.  Discard previously
-         * found file names and start all over again.
-         */
-        else if (has_env_var(p) || *p == '~') {
+        if (has_env_var(p) || *p == '~') {
+          // If expand_env() can't expand an environment
+          // variable, use the shell to do that.  Discard previously
+          // found file names and start all over again.
```

Comment:
Keeping this "On Unix" does look clearer.
```suggestion
        // On Unix, if expand_env() can't expand an environment
        // variable, use the shell to do that.  Discard previously
        // found file names and start all over again.
        if (has_env_var(p) || *p == '~') {
```